### PR TITLE
24378 - script to add Registrar's Notation to all existing companies

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#24378

*Description of changes:*

Jupyter notebook to add Registrar's Notation to all existing companies.

I tried for some BENs in DEV. See below screenshot

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/notebook/add_registrars_notation.ipynb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    "                \"name\": \"registrarsNotation\",\n",
+    "                \"date\": str(current_date),\n",
+    "                \"certifiedBy\": \"system\"\n",
+    "            },\n",
+    "            \"business\": {\n",
+    "                \"identifier\": ben,\n",
+    "                \"legalType\": \"BEN\"\n",
+    "            },\n",
+    "            \"registrarsNotation\": {\n",
+    "                \"orderDetails\": \"This is a test to add Registrars Notation \"\n",
```
Comment (vysakh-menon-aot) [inline legal-api/notebook/add_registrars_notation.ipynb:91]: Let's update it before merging this PR

File: legal-api/notebook/add_registrars_notation.ipynb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    "    'Authorization': 'Bearer ' + token\n",
+    "}\n",
+    "\n",
+    "# loop through list of businesses to create filing\n",
+    "for ben in ben_businesses:\n",
+    "    filing_data = {\n",
+    "        \"filing\": {\n",
+    "            \"header\": {\n",
+    "                \"name\": \"registrarsNotation\",\n",
+    "                \"date\": str(current_date),\n",
```
Comment (vysakh-menon-aot) [inline legal-api/notebook/add_registrars_notation.ipynb:99]: Its already string now
```suggestion
    "                \"date\": current_date,\n",
```

**Review by vysakh-menon-aot — APPROVED** [review]

**Comment by vysakh-menon-aot** [comment]
@ketaki-deodhar I missed this one. Can you move this into `lear/jobs` by creating new folder (`correction-ben-statement`)? legal-api is not the right place for this notebook

**Comment by ketaki-deodhar** [comment]
> @ketaki-deodhar I missed this one. Can you move this into `lear/jobs` by creating new folder (`correction-ben-statement`)? legal-api is not the right place for this notebook

Sure. Will do

File: jobs/correction-ben-statement/data.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "BC0871062",
+    "BC0871261",
+    "BC0871183",
+    "BC0871181",
+    "BC0871152",
+    "BC0871062",
+    "BC0871139",
+    "BC0871073",
+    "BC0870819",
+    "BC0870818"
```
Comment (severinbeauvais) [inline jobs/correction-ben-statement/data.py:14]: Is this the list of businesses in Prod?

If not, when do we update and commit this file?

Comment (severinbeauvais) [reply jobs/correction-ben-statement/data.py:14]: Since this is a committed file, how can we test this in Test environment?

Comment (ketaki-deodhar) [reply jobs/correction-ben-statement/data.py:14]: This is not a list from PROD. I think we will have to get the list from PROD around/closer to the deployment date, dump the list here and then run script.
Yes. I think we can test this in TEST once released to test. We just need to get the test data (identifiers).
@vysakh-menon-aot please add if I am missing anything

Comment (vysakh-menon-aot) [reply jobs/correction-ben-statement/data.py:14]: Consider this as a one time script, only difference is its a jupiter notebook. We are not deploying this job in openshift. We have to manually execute this in each environment with the list of identifier.

Comment (ketaki-deodhar) [reply jobs/correction-ben-statement/data.py:14]: yes. I will add all the details in the ticket

Comment (vysakh-menon-aot) [reply jobs/correction-ben-statement/data.py:14]: @ketaki-deodhar its better not to keep all these identifiers in repo. Keep 1 or 2 as a sample and remove rest of it

**Comment by ketaki-deodhar** [comment]
Example of Registrar's Notation added in DEV:

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**